pkg/updater/updater.go

```diff
@@ -109,8 +115,56 @@ func lockGroup(ctx context.Context, client gcs.ConditionalClient, path gcs.Path,
 	return gcs.Touch(ctx, client, path, generation, buf)
 }
 
+func update(ctx context.Context, client gcs.ConditionalClient, mets Metrics, log logrus.FieldLogger, tg *configpb.TestGroup, configPath gcs.Path, gridPrefix string, updateGroup GroupUpdater, write bool, generations map[string]int64) {
+	var errored bool
+	defer func() {
+		if errored {
+			if mets.Errors != nil {
+				mets.Errors.Add(1, "updater")
```

Comment:
Seems kind of strange to make the caller remember to add an updater string. Seems like this should be embedded into either the mets or Errors structure.